compiler-plugin/src/main/java/io/ballerina/lib/data/jsondata/compiler/JsondataTypeValidator.java

```diff
@@ -155,22 +156,31 @@ private void processFunctionDefinitionNode(FunctionDefinitionNode functionDefini
     }
 
     private void checkTypeAndDetectDuplicateFields(TypeSymbol typeSymbol, SyntaxNodeAnalysisContext ctx) {
+        checkTypeAndDetectDuplicateFields(typeSymbol, ctx, new HashSet<>());
+    }
+
+    private void checkTypeAndDetectDuplicateFields(TypeSymbol typeSymbol,
```

Comment:
Shall we have tests for these scenarios?

Reply:
There are already tests regarding this. I added another one.